Benjamin - Benjamin-Tu/bt_week4_assignment.do

```diff
@@ -0,0 +1,188 @@
+
+
+global wd "/Users/benjamintu/Downloads/Econ490_Fall2022/Week4/04_assignment/data"
+
+use "$wd/village_pixel.dta", clear
+
+* Q1 a)
+egen pixel_mean = mean(payout),by(pixel) //calculate the mean for each different pixel.
+
+gen pixel_consistent = 0 if pixel_mean== 0 | pixel_mean ==1    // pixel is consistent if the mean is an integer.
+
+replace pixel_consistent = 1 if pixel_mean!= 0 & pixel_mean!= 1 // if the mean is not an integer, pixel is not consistent
+
+* b)
+/*
+why the value changes when I try to destring?
+gen nu_pixel = substr(pixel,3,4)
+destring (nu_pixel), replace
+egen nu_pixel_mean = mean(nu_pixel),by(village)
+*/
+
+sort village
+
+egen tag_pixel_village = tag(village pixel) 
+
+egen tag_pixel_village_mean = sum(tag_pixel_village),by(village)
+
+gen pixel_village=0 if  tag_pixel_village_mean == 1
+
+replace pixel_village=1 if tag_pixel_village_mean >1
+
+* c)
+gen village_categories=1 if pixel_village==0 
+
+replace village_categories=2 if pixel_village==1 & pixel_consistent==0
+
+replace village_categories=3 if pixel_village==1 & pixel_consistent==1
+
+tab hhid if village_categories==2
```

Comment:
When I ran this section, my `village_categories` variable only had two values: 1 and 2 (not 3). I think the issue is that you had (and used) a variable indicating if the payout was consistent within a pixel, but not if the payout is consistent within a _village_, as the question asks. Therefore, you need to create a new variable that is 1 if the _village_ (not the pixel) has different payout status and 0 if it does not. Once you've created this variable, you can use it (and `pixel_village`) to create the `village_categories` variable.

See relevant portion of question in bold here: "For this experiment, it is only an issue **if villages are in different pixels AND have different payout status**"